distribution/std-lib/Base/src/Number_Extensions.enso

```diff
@@ -0,0 +1,85 @@
+from Base import all
+polyglot java import java.lang.Math
+polyglot java import java.lang.String
+
+## Computes the inverse of the sine function
+
+   Selects a value in the -pi/2 through pi/2 range.
+Number.asin : Decimal
+Number.asin = Math.asin [this.to_decimal]
+
+## Computes the inverse of the cosine function.
+
+   Selects a value in the -pi/2 through pi/2 range.
+Number.acos : Decimal
+Number.acos = Math.acos [this.to_decimal]
+
+## Computes the inverse of the tangent function.
+
+   Selects a value in the -pi/2 through pi/2 range.
+Number.atan : Decimal
+Number.atan = Math.atan [this.to_decimal]
+
+## Computes the argument (angle) in the conversion from cartesian
+   to polar coordinates.
+
+   The returned angle is in the -pi through pi range.
+Number.atan_2 : Number -> Decimal
+Number.atan_2 y = Math.atan2 [this.to_decimal, y.to_decimal]
+
+## Computes the sine function.
+Number.sin : Decimal
+Number.sin = Math.sin [this.to_decimal]
+
+## Computes the cosine function.
+Number.cos : Decimal
+Number.cos = Math.cos [this.to_decimal]
+
+## Computes the tangent function.
+Number.tan : Decimal
+Number.tan = Math.tan [this.to_decimal]
+
+## Computes the hyperbolic sine function.
+Number.sinh : Decimal
+Number.sinh = Math.sinh [this.to_decimal]
+
+## Computes the hyperbolic cosine function.
+Number.cosh : Decimal
+Number.cosh = Math.cosh [this.to_decimal]
+
+## Computes the hyperbolic tangent function.
+Number.tanh : Decimal
+Number.tanh = Math.tanh [this.to_decimal]
+
+## Computes the exponential function.
```

Comment:
Please have a todo comment for implementing this with an operator `^`.

Reply:
Nope :P There is `pow` in `^`, this is a helper computing `e ^ x`.